File: Arcade/Intro/Java/ReverseInParentheses.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

final class ReverseInParentheses {

    String reverseInParentheses(String inputString) {
        Matcher mt = Pattern.compile("\\(\\w*\\)").matcher(inputString);
        if(!mt.find()) return inputString;
        String str = mt.group(),
            revStr = str.replace(")", "").replace("(", "");
        return reverseInParentheses(inputString.replace(str,
            new StringBuilder(revStr).reverse().toString()));
    }
}

File: Arcade/Intro/javascript/reverseInParentheses.js
function reverseString(str) {
    let o = "";
    for(let i = str.length - 1; i >= 0; i--) o += str.charAt(i);
    return o;
}

function reverseInParentheses(inputString) {
    let reg = /\(\w*\)/;
    if(!reg.test(inputString)) return inputString;
    let str = reg.exec(inputString)[0],
        revStr = reverseString(str.substr(1, str.length - 2));
    return reverseInParentheses(inputString.replace(str, revStr));
}